dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeVectorStoreRecordCollection.cs

```diff
@@ -90,61 +79,100 @@ public PineconeVectorStoreRecordCollection(Sdk.PineconeClient pineconeClient, st
-                var collections = await this._pineconeClient.ListIndexes(cancellationToken).ConfigureAwait(false);
+                var collections = await this._pineconeClient.ListIndexesAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
 
-                return collections.Any(x => x.Name == this.CollectionName);
-            }).ConfigureAwait(false);
-
-        return result;
-    }
+                return collections.Indexes?.Any(x => x.Name == this.CollectionName) is true;
+            });
 
     /// <inheritdoc />
-    public virtual async Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    public virtual Task CreateCollectionAsync(CancellationToken cancellationToken = default)
     {
         // we already run through record property validation, so a single VectorStoreRecordVectorProperty is guaranteed.
         var vectorProperty = this._propertyReader.VectorProperty!;
-        var (dimension, metric) = PineconeVectorStoreCollectionCreateMapping.MapServerlessIndex(vectorProperty);
 
-        await this.RunOperationAsync(
-            CreateCollectionName,
-            () => this._pineconeClient.CreateServerlessIndex(
-                this.CollectionName,
-                dimension,
-                metric,
-                this._options.ServerlessIndexCloud,
-                this._options.ServerlessIndexRegion,
-                cancellationToken)).ConfigureAwait(false);
+        CreateIndexRequest request = new()
+        {
+            Name = this.CollectionName,
+            Dimension = vectorProperty.Dimensions ?? throw new InvalidOperationException($"Property {nameof(vectorProperty.Dimensions)} on {nameof(VectorStoreRecordVectorProperty)} '{vectorProperty.DataModelPropertyName}' must be set to a positive integer to create a collection."),
+            Metric = MapDistanceFunction(vectorProperty),
+            Spec = new ServerlessIndexSpec
+            {
+                Serverless = new ServerlessSpec
+                {
+                    Cloud = MapCloud(this._options.ServerlessIndexCloud),
+                    Region = this._options.ServerlessIndexRegion,
+                }
+            },
+        };
+
+        return this.RunOperationAsync("CreateCollection",
+            () => this._pineconeClient.CreateIndexAsync(request, cancellationToken: cancellationToken));
     }
 
     /// <inheritdoc />
     public virtual async Task CreateCollectionIfNotExistsAsync(CancellationToken cancellationToken = default)
     {
         if (!await this.CollectionExistsAsync(cancellationToken).ConfigureAwait(false))
         {
-            await this.CreateCollectionAsync(cancellationToken).ConfigureAwait(false);
+            try
+            {
+                await this.CreateCollectionAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (VectorStoreOperationException ex) when (ex.InnerException is PineconeApiException apiEx && apiEx.InnerException is ConflictError)
```

Comment:
I know you're not a fan of pattern matching but:

```suggestion
            catch (VectorStoreOperationException ex) when (ex.InnerException is PineconeApiException { InnerException: ConflictError })
```